includes/plugins/class-co-authors-plus.php

```diff
@@ -103,5 +170,228 @@ public static function setup_custom_role_and_capability() {
 
 		\update_option( self::SETTINGS_VERSION_OPTION_NAME, $current_settings_version );
 	}
+
+		/**
+		 * Filters user validation to allow empty emails for guest authors
+		 *
+		 * When creating a new user, also automatically generate a username from the display name.
+		 *
+		 * @param WP_Error $errors WP_Error object (passed by reference).
+		 * @param bool     $update Whether this is a user update.
+		 * @param stdClass $user   User object (passed by reference).
+		 * @return WP_Error
+		 */
+	public static function user_profile_update_errors( $errors, $update, $user ) {
+
+		if ( self::CONTRIBUTOR_NO_EDIT_ROLE_NAME !== $user->role ) {
+			return $errors;
+		}
+
+		if ( ! empty( $errors->errors['empty_email'] ) ) {
+			$errors->remove( 'empty_email' );
```

Comment:
Done in https://github.com/Automattic/newspack-plugin/pull/3277/commits/d8848d3b066ebdd984af7afdda7a94397fe9e91b

This will most certainly be needed for Newspack Network, since we use the email as the user id across the network. thanks!